### DISCUSSION TITLE
MEGA65: ideas / wish-list / suggestions

### DISCUSSION THREAD

**Discussion by lgblgblgb** [lgblgblgb/xclcd#324] [Ideas]
## Ideas / wish-list / suggestions on MEGA65 emulator of Xemu

Note, there is no guarantee these will be implemented and especially no time-frame for implementing these. But it's important to collect these kind of information to have some priority list and plan the development and so on.

**Comment by skeetor** [comment]
A debugger interface similar to VICE would be cool.

**Reply by lgblgblgb** [reply]
Thanks for your suggestion! That functionality is already there, well, "basically" :) The design of this is to mimic the original MEGA65, where you can attach external debugger, namely `m65dbg`. It works with Xemu too, in fact, the goal was to use the same tools with both of real and emulated machine. OK, I admit, the Xemu implementation of that debug protocol is not so great at the moment and have "some issues". Surely it still needs a _tons_ of work to be good, but in this case we must take care that it has to be compatible with the protocol used on real MEGA65 with the same purpose (using a serial-over-USB connection in that case, on Xemu it's a TCP/IP socket, but the protocol itself is the same).
